src/MongoDB/ReadPreference.c

```diff
@@ -429,14 +433,29 @@ static HashTable* php_phongo_readpreference_get_properties_hash(zval* object, bo
 		php_phongo_bson_state state = PHONGO_BSON_STATE_INITIALIZER;
 		state.map.root_type         = PHONGO_TYPEMAP_NATIVE_ARRAY;
 
-		php_phongo_bson_to_zval_ex(bson_get_data(tags), tags->len, &state);
+		if (php_phongo_bson_to_zval_ex(bson_get_data(tags), tags->len, &state)) {
+#if PHP_VERSION_ID >= 70000
+			Z_ADDREF(state.zchild);
```

Comment:
I'm kind of curious as to whether there is a reason we `Z_ADDREF` here and always `zval_ptr_dtor` after updating the hash table (for either PHP version). This seems a bit different than earlier functions I reviewed in the PR.

AFAIK, `Z_ADDREF` and `zval_ptr_dtor` simply cancel out. Instead, I would think we'd simply update the hash table if BSON conversion succeeds, and call `zval_ptr_dtor` for the `else` (i.e. failure) case. Do you think it's worth changing this, or perhaps putting off the possible refactoring to another ticket? If you'd like to test this for leaks, you can run the test suite (or a portion) with `TEST_PHP_ARGS=-m` defined to enable Valgrind.

Reply:
You are correct. All other instances where we add a zval to the response simply don't call `zval_ptr_dtor` instead of adding to the refcount. I've updated this block to be consistent with other calls.

There are some more occurrences regarding `Z_ADDREF` when preparing the scope in `MongoDB\BSON\Javascript` instances for output. Given that #1036 changes the point in time when we validate scope, I've created [PHPC-1471](https://jira.mongodb.org/browse/PHPC-1471) to take care of this later and added the respective tickets as dependencies.